<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>data_preprocessing</class>
 <widget class="QDialog" name="data_preprocessing">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>518</width>
    <height>739</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Data Preprocessing</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background: None;</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <spacer name="verticalSpacer_3">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QGroupBox" name="target_label_groupBox">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>50</height>
      </size>
     </property>
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;the class variable you want to predict&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="title">
      <string>Target Label</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="3">
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="2">
       <widget class="QComboBox" name="target_index_comb"/>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="set_target_cb">
        <property name="text">
         <string>response variable at index of </string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_4">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QGroupBox" name="text_feature_groupBox">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>100</height>
      </size>
     </property>
     <property name="title">
      <string>Text Feature</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <property name="verticalSpacing">
       <number>9</number>
      </property>
      <item row="1" column="0">
       <widget class="QCheckBox" name="drop_feature_by_index_cb">
        <property name="text">
         <string>drop feature at column index of </string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLineEdit" name="drop_index_le">
        <property name="text">
         <string>0,482</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Feature index from 0</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLineEdit" name="onehot_index_le"/>
      </item>
      <item row="2" column="3">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="trans_into_ordinal_cb">
        <property name="text">
         <string>transform the text feature into ordinal</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLineEdit" name="ordinal_index_le"/>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="trans_into_onehot_cb">
        <property name="text">
         <string>transform the text feature into one-hot</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QGroupBox" name="scale_groupBox">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>50</height>
      </size>
     </property>
     <property name="title">
      <string>Scale Features ( Except the target feature )</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="0" column="0">
       <widget class="QCheckBox" name="scaled_cb">
        <property name="text">
         <string>features scaled by</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QComboBox" name="scale_by_comb">
        <item>
         <property name="text">
          <string>StandardScaler</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>MinMaxScaler</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>RobustScaler</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="1">
       <spacer name="horizontalSpacer_7">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="3">
       <spacer name="horizontalSpacer_8">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QGroupBox" name="table_groupBox">
     <property name="minimumSize">
      <size>
       <width>500</width>
       <height>250</height>
      </size>
     </property>
     <property name="title">
      <string>Preview *unprocessed</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QTableWidget" name="tableWidget"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="widget" native="true">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>50</height>
      </size>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="horizontalSpacing">
       <number>20</number>
      </property>
      <item row="0" column="0">
       <widget class="QPushButton" name="apply_btn">
        <property name="text">
         <string>Apply</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="finish_btn">
        <property name="text">
         <string>Finish</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="reset_btn">
        <property name="text">
         <string>Reset Data From Previous</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>drop_feature_by_index_cb</sender>
   <signal>clicked(bool)</signal>
   <receiver>data_preprocessing</receiver>
   <slot>parameter_changed_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>256</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>162</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trans_into_ordinal_cb</sender>
   <signal>clicked(bool)</signal>
   <receiver>data_preprocessing</receiver>
   <slot>parameter_changed_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>248</y>
    </hint>
    <hint type="destinationlabel">
     <x>191</x>
     <y>180</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trans_into_onehot_cb</sender>
   <signal>clicked(bool)</signal>
   <receiver>data_preprocessing</receiver>
   <slot>parameter_changed_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>206</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>438</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>drop_index_le</sender>
   <signal>textEdited(QString)</signal>
   <receiver>data_preprocessing</receiver>
   <slot>parameter_changed_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>455</x>
     <y>221</y>
    </hint>
    <hint type="destinationlabel">
     <x>384</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ordinal_index_le</sender>
   <signal>textEdited(QString)</signal>
   <receiver>data_preprocessing</receiver>
   <slot>parameter_changed_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>455</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>444</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>onehot_index_le</sender>
   <signal>textEdited(QString)</signal>
   <receiver>data_preprocessing</receiver>
   <slot>parameter_changed_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>455</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>409</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>set_target_cb</sender>
   <signal>clicked(bool)</signal>
   <receiver>data_preprocessing</receiver>
   <slot>parameter_changed_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>436</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>target_index_comb</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>data_preprocessing</receiver>
   <slot>parameter_changed_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>397</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>457</x>
     <y>431</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>finish_btn</sender>
   <signal>clicked()</signal>
   <receiver>data_preprocessing</receiver>
   <slot>finish_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>499</x>
     <y>715</y>
    </hint>
    <hint type="destinationlabel">
     <x>360</x>
     <y>678</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>apply_btn</sender>
   <signal>clicked()</signal>
   <receiver>data_preprocessing</receiver>
   <slot>apply_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>715</y>
    </hint>
    <hint type="destinationlabel">
     <x>68</x>
     <y>681</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scaled_cb</sender>
   <signal>clicked(bool)</signal>
   <receiver>data_preprocessing</receiver>
   <slot>parameter_changed_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>381</y>
    </hint>
    <hint type="destinationlabel">
     <x>87</x>
     <y>450</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scale_by_comb</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>data_preprocessing</receiver>
   <slot>parameter_changed_handler()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>382</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>345</x>
     <y>454</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>parameter_changed_handler()</slot>
  <slot>apply_handler()</slot>
  <slot>finish_handler()</slot>
 </slots>
</ui>
